src/xenia/cpu/xex_module.cc

```diff
@@ -41,11 +41,19 @@ XexModule::XexModule(Processor* processor, KernelState* kernel_state)
       processor_(processor),
       kernel_state_(kernel_state),
       xex_(nullptr),
+      xex_header_(nullptr),
       base_address_(0),
       low_address_(0),
       high_address_(0) {}
 
-XexModule::~XexModule() { xe_xex2_dealloc(xex_); }
+XexModule::~XexModule() {
+  xe_xex2_dealloc(xex_);
+
+  if (xex_header_) {
+    delete[] xex_header_;
```

Comment:
Not messy - quite the opposite - no need for this block of code at all, no need for nullptr init, and the malloc is handled under the covers. In general, anything we can do to remove sources of leaks, uninitialized memory, etc is a win. It's kind of what STL is for :) If we find someplace where the STL type isn't appropriate for some reason we should then at least roll our own that has the same properties, however *that's* what leads to confusion. It's like using std::string instead of naked char*.